Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js

```diff
@@ -2,6 +2,32 @@ import { performSearch } from "../../utilities/supportUtilities";
 import { getTableRows } from "../../utilities/sharedUtilities";
 
 const hdid = "P6FFO433A5WPMVTGM7T4ZVWBKCSVNAYGTWTU3J2LWMGUMERKI72A";
+const switchName = "Immunization";
+const auditBlockReason = "Test block reason";
+const auditUnblockReason = "Test unblock reason";
+
+function checkAgentAuditHistory(expectedRows = null) {
+    cy.get("[data-testid=agent-audit-history-title]")
+        .should("be.visible")
+        .click();
+
+    cy.get("[data-testid=agent-audit-history-table]").should("be.visible");
+
+    let historyCount = 0;
+    return new Promise((resolve, reject) => {
+        cy.get("[data-testid=agent-audit-history-title]")
+            .invoke("text")
+            .then((text) => {
+                const regex = /\((\d+)\)/g;
+                const match = regex.exec(text);
+                historyCount = Number(match[1]);
+                if (expectedRows != null && expectedRows >= 0) {
+                    expect(historyCount).to.equal(expectedRows);
+                }
+                resolve(historyCount);
+            });
+    });
```

Comment:
I am not sure about this type of complexity.  @MikeLyttle  what do you think?

Reply:
great this is a good suggestion. I'll simplify

Reply:
I've simplified the approach, as its in a table it was easiest just to put the count in the span.